refgenconf/refgenconf.py

```diff
@@ -2087,10 +2162,134 @@ def _select_genomes(self, genome=None, strict=False, order=None,
         return genomes if not all(x in genomes for x in genome) else genome
 
+    def config_upgrade(target_version, filepath, force=False):
+        """
+            upgrade the config file to a target version, 
+            and alter genome_folder stucture 
+
+            :param str target_version: the version updated to
+            :param str filepath: file path for the current config file
+            :param bool force
+        """
+
+        # check if any genome lack of local fasta asset and not on server
+
+        def check_genome_digests(config):
+            missing_digest = []
+            for k, v in config["genomes"].items():
+                if not os.path.exists(config["genome_folder"]+'/'+k +
+                                      '/fasta/default/'+k+'.fa'):
+                    response = requests.get(
+                        "http://refgenomes.databio.org:82/v3/alias/genome_digest/"+k)
+                    if response.json()['detail']:
+                        missing_digest.append(k)
+
+            if missing_digest and not query_yes_no(
+                    "The following genome(s) would be lost due to the lack of local and remote fasta asset(s): \n"
+                    "{} \n"
+                    "Would you like to proceed?"
+                    .format(missing_digest)):
+                _LOGGER.info("Action aborted by the user. To use it, please downgrade refgenie: 'pip install \"refgenie>={},<{}\"'".
+                             format(REFGENIE_BY_CFG[str(config["config_version"])], REFGENIE_BY_CFG[str(REQ_CFG_VERSION)]))
+                return missing_digest
+        # reformat config file
+
+        def format_config(config, target_version):
+            # reformat the config file
+            for k, v in config["genomes"].items():
+                # create "aliases" section
+                v.aliases = [k]
+                # get genome digest from the server
+                response = requests.get(
+                    "http://refgenomes.databio.org:82/v3/alias/genome_digest/"+k)
+                if response.json()['detail']:
+                    # if genome asset not exist on serer, generate the digest using the local fasta asset
+                    ssc = SeqColClient({})
+                    try:
+                        d, _ = ssc.load_fasta(config["genome_folder"]+'/'+k +
+                                              '/fasta/default/'+k+'.fa')
+                    except OSError:
+                        del config["genomes"][k]
+                        continue
+                else:
+                    d = response.json()
+                # convert seek keys, childran/parent asset keys from aliases to genome digests
+                for asset_k, asset_v in v["assets"].items():
+                    for seek_k, seek_v in asset_v["tags"]["default"]["seek_keys"].items():
+                        asset_v["tags"]["default"]["seek_keys"][seek_k] = seek_v.replace(
```

Comment:
This is not gonna work; the `tag` value is hard coded (`"default"`). You need to account for multiple tags and not assume `"default"` is defined.

Reply:
64f0ca91f0b5e150ee86189b77166e952d7287bb